Assets/Scripts/Monitor.cs

```diff
@@ -111,7 +111,8 @@ private void RecordTelemetry() {
       Vector3 vel = agent.GetVelocity();  // Ensure GetVelocity() doesn't allocate
 
       int agentID = agent.GetInstanceID();
-      int flightPhase = (int)agent.GetFlightPhase();
+      int flightPhase =
+          agent is AirborneAgent airborneAgent ? (int)airborneAgent.GetFlightPhase() : -1;
```

Comment:
```suggestion
          (int)(agent is AirborneAgent airborneAgent ? airborneAgent.GetFlightPhase() : FlightPhase.INVALID);
```
and in the enum, I would add an `INVALID = -1`